lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -494,6 +494,17 @@ function validatePolicyResource(bucketName, policy) {
     });
 }
 
+// Apply the existing IP validation logic to each element
+function validateIpRegex(ip) {
+    if (constants.ipv4Regex.test(ip)) {
+        return ip.split('.').every(part => parseInt(part, 10) <= 255);
+    }
+    if (constants.ipv6Regex.test(ip)) {
+        return ip.split(':').every(part => part.length <= 4);
+    }
+    return false;
+}
```

Comment:
We should unit test it when it's done

Reply:
This was already there I just moved the function out of the loop to avoid recreating it multiple time

Reply:
Yes, I meant, now that it's not inline but a function, we can test it independently :slightly_smiling_face:

Reply:
Dropped and will be done in ticket https://scality.atlassian.net/browse/CLDSRV-627